src/main/kotlin/io/zerobase/smarttracing/resources/OrganizationsResource.kt

```diff
@@ -111,16 +121,14 @@ class OrganizationsResource(val dao: GraphDao, private val siteTypes: MultiMap<S
             )
         }
 
-        return when (id) {
-            null -> null
-            else -> IdWrapper(id)
-        }
+        return id?.let { IdWrapper(id) }
     }
 
     @Path("/{id}/sites")
     @GET
-    fun getSites(@PathParam("id") id: String): List<Any> {
-        return dao.getSites(id)!!
+    fun getSites(@PathParam("id") id: String): List<SiteResponse> {
+        val (x, y) = dao.getSites(id).unzip()
+        return x.zip(y) { a:String, b:String -> SiteResponse(a, b) }
```

Comment:
nit: slightly simpler: `return dao.getSites(id).let { (a,b) -> SiteResponse(a, b) }` or `return dao.getSites(id).let{ SiteResponse(it.first, it.second) }`

Reply:
tried both those, was throwing questionable cases, will try again but i think it will still have that issue still

Reply:
ya let will not work, expected type mismatch.
using map will result in the questionable cast, to prevent multiple suppressions i just did it in a way without requiring to suppress a warning